app/src/main/java/io/capibaras/abcall/viewmodels/SignUpViewModel.kt

```
@@ -122,65 +117,52 @@ class SignUpViewModel(
 
     private fun getCompanies(forceUpdate: Boolean = false) {
         viewModelScope.launch {
-            isLoading = true
+            sharedViewModel.setLoadingState(true)
             try {
                 companies = companyRepository.getCompanies(forceUpdate)
             } catch (e: IOException) {
-                errorUIState = ErrorUIState.Error(R.string.error_network)
+                sharedViewModel.setErrorState(ErrorUIState.Error(R.string.error_network))
             } catch (e: Exception) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)